# Importing Library

In [2]:
# import libraries

import pandas as pd
import numpy as np
import os

# Path

In [3]:
#Creating a path varaible - Shortcut for Importing Data Files
path = r'C:\Users\Mallika\Documents\Career Foundry\Module2- Data Immersion\Achievement 4\June2025_Instacart Basket Analysis'

# 1. Importing Data

In [4]:
# importing dataframe

df_ords = pd.read_csv(os.path.join(path, '02 data' , 'prepared data', 'orders_checked.csv'), index_col=0)

df_ords_prior = pd.read_csv(os.path.join(path, '02 data' , 'original data' , 'orders_products_prior.csv'))

# 2. Dimension of the imported Dataframe

In [5]:
# Check the output
df_ords_prior.head()

order_id  product_id  add_to_cart_order  reordered
0         2       33120                  1          1
1         2       28985                  2          1
2         2        9327                  3          0
3         2       45918                  4          1
4         2       30035                  5          0

In [6]:
df_ords.head()

order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   2539329        1             1          2                  8   
1   2398795        1             2          3                  7   
2    473747        1             3          3                 12   
3   2254736        1             4          4                  7   
4    431534        1             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

In [9]:
# Check the size

df_ords_prior.shape

(32434489, 4)

In [7]:
df_ords.shape

(3421083, 6)

# 3.Data Merge

In [10]:
# inner merge dataframe 

df_merge_large = df_ords.merge(df_ords_prior, on = ['order_id'] , how = 'inner', indicator = True)

In [11]:
# examine value counts on '_merge' column

df_merge_large['_merge'].value_counts()

_merge
both          32434489
left_only            0
right_only           0
Name: count, dtype: int64

Observation: The resulting dataframe (after the merge) has 32,434,489 rows, and each of those rows have information found in both input data sets, as we used an inner join for the purposes of this project.

# 4.Exporting data as pickle

In [12]:
df_merge_large.to_pickle(os.path.join(path, '02 data' , 'prepared data' , 'orders_products_combined.pkl'))